corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -434,38 +424,61 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
             var self = this;
             e.preventDefault();
 
-            var invalidFields = [];
-            var errorHTML = gettext("Please check the following fields:");
+            // validateFields will likely already have been called when user blurred the last field,
+            // but call it here just in case they didn't fill anything out
+            self.validateFields().done(function () {
+                FormplayerFrontend.trigger("menu:query", self.getAnswers());
+            });
+        },
 
-            FormplayerFrontend.trigger('clearNotifications', errorHTML, true);
+        /*
+         *  Send request to formplayer to validate fields. Displays any errors.
+         *  Returns a promise that contains the formplayer response.
+         */
+        validateFields: function () {
```

Comment:
Are you worried about this being called much more often than it needs to be?  I wonder if it'd be worth trying to figure out some logic around caching this and expiring when user input changes.  Sounds tricky.

Reply:
I'm a little bit worried about it. I'm planning to put that worry out of my head until we get complaints (acknowledging that might not take long).